lsp-mode.el

```diff
@@ -2887,26 +2887,20 @@ interface Range {
               (not (gethash "textDocument" it))
               (let* ((ident (gethash "textDocument" it))
                      (filename (lsp--uri-to-path (gethash "uri" ident)))
-                     (version (gethash "version" ident)))
-                (with-current-buffer (find-file-noselect filename)
-                  (or (null version) (zerop version)
+                     (version (gethash "version" ident))
+                     (buf (lsp--buffer-for-file filename)))
+                (or (null version) (zerop version) (null buf)
+                    (with-current-buffer buf
                       (equal version lsp--cur-version))))))
            document-changes)
     (error "Document changes cannot be applied")))
 
 (defun lsp--apply-workspace-edit (edit)
   "Apply the WorkspaceEdit object EDIT."
-  (if-let (document-changes (seq-reverse (gethash "documentChanges" edit)))
+  (if-let (document-changes (gethash "documentChanges" edit))
       (progn
         (lsp--check-document-changes-version document-changes)
-        (->> document-changes
```

Comment:
Some of the language servers sent `(Rename (Apply Changes To Original File))` while the other sent `((Apply Changes To Original File) Rename)` . One of the servers was jdtls(but I am not sure what was the other).

Reply:
It would be good if there was more information, so that we could at least file a bug against the non-conforming LSP server.

I'm looking at the jdt.ls `ChangeUtil.java` and at least here it is putting edits before the rename:

https://github.com/eclipse/eclipse.jdt.ls/blob/258729fbf2e54b64e88dfa489f3d5133d882b695/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ChangeUtil.java#L202

and

https://github.com/eclipse/eclipse.jdt.ls/blob/258729fbf2e54b64e88dfa489f3d5133d882b695/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ChangeUtil.java#L210-L212

but there are several places where instances of `RenameFile` are created.

Reply:
I will look for the what was the server sending odd edits, I spent some time looking into the git log and the initial implementation didn't have the reverse piece. I am primary using jdtls and IIRC it was working fine.

Then, I found a server that might send several workspace edit updates for a single file, unfortunately, I haven't included what was the server in the commit message. I will try to find out what was the server doing that.